package.json

```diff
@@ -18,30 +22,31 @@
     "homepage": "https://github.com/GPII/gpii-express-user",
     "dependencies": {
         "foundation-sites": "6.3.1",
-        "gpii-binder": "1.0.3",
-        "gpii-express": "1.0.14",
-        "gpii-handlebars": "1.1.0",
-        "gpii-json-schema": "1.0.5",
-        "infusion": "3.0.0-dev.20180326T173646Z.8c6a109b1",
-        "kettle": "1.7.3",
-        "markdown-it": "8.4.1",
-        "nodemailer": "4.6.7",
+        "gpii-binder": "1.1.0",
+        "gpii-express": "1.0.15",
+        "gpii-handlebars": "2.1.0",
+        "gpii-json-schema": "2.1.2",
+        "infusion": "3.0.0-dev.20191009T141140Z.32c9263b4.FLUID-6148",
```

Comment:
Please try for 3.0.0-dev.20200317T122331Z.1aa7ea8a1.FLUID-6145 if possible

Reply:
This catastrophically breaks things with complaints about circular references.  I will have to give it a go tomorrow.

Reply:
```
19:34:16.706:  ASSERTION FAILED: Error in component configuration - a circular reference was found during evaluation of path segment "schema of path [
    "options",
    "schema",
    "definitions",
    "username"
]": for more details, see the activity records following this message in the console, or issue fluid.setLogging(fluid.logLevel.TRACE) when running your application
19:34:16.707:  Current activity: 
19:34:16.707:      while resolving context value {
    "context": "that",
    "path": "options.schema.definitions.username",
    "endpos": 42
}

```

Looking at [the grade in question](https://github.com/the-t-in-rtf/gpii-express-user/blob/836e94c126f1e766d81b6035ce745174adf026cf/src/js/common/schemaHolders.js#L118), I don't see how this is circular.  It's the express strategy we've discussed for (re)using JSON schema-like "definitions" using IoC instead of their inheritance strategy.

Reply:
Thanks, this is a helpful report, if a very unwelcome one - it looks like FLUID-6145 contains an additional "retrunking" bug of the kind tracked in https://issues.fluidproject.org/browse/FLUID-4930 . I had encountered this in my own work and written a failing test case that looks so https://github.com/amb26/infusion/compare/FLUID-6145...amb26:FLUID-4930#diff-ab93103a69113cb998d7630a3c4f4451R506 - I thought I could dodge fixing it but given you have what looks like a pretty vanilla triggering of it there seems to be little alternative but to write it up and fix it. I had thought that the use of an expander was essential to triggering the bug but it seems not.

Reply:
The diff link doesn't show me anything?

Reply:
Looks like the markdown renderer has stuffed up the anchor - here it is without: https://github.com/amb26/infusion/compare/FLUID-6145...amb26:FLUID-4930

Reply:
@amb26, I see the circular dependency detection issues when running the tests locally using either FLUID-6145 or FLUID-6148.  The tests in gpii-json-schema itself work fine with FLUID-6148, I suspect that's because I don't have tests for complex reuse of material via IoC in that package (I'll add some).

Is there a branch that includes the fixes for [FLUID-6503](https://issues.fluidproject.org/browse/FLUID-6503) but which does not include the circular dependency checks tripping this up (or that includes a fix)?

I need to be able to reliably remix schema material like this using IoC in the LSR era, so any help you can provide would be helpful.